pkg/cloud/azureprovider.go

```diff
@@ -734,6 +828,49 @@ func (az *Azure) AllNodePricing() (interface{}, error) {
 func (az *Azure) NodePricing(key Key) (*Node, error) {
 	az.DownloadPricingDataLock.RLock()
 	defer az.DownloadPricingDataLock.RUnlock()
+	config, _ := az.Config.GetCustomPricingData()
+	if config.SpotLabel != "" && config.SpotLabelValue != "" {
+		labelsMapStr := fmt.Sprintf("%s", key)
+		spotLabelMapStr := " " + config.SpotLabel + ":" + config.SpotLabelValue + " "
+		if strings.Contains(labelsMapStr, spotLabelMapStr) {
+			features := strings.Split(key.Features(), ",")
+			spotFeatures := strings.ReplaceAll(key.Features(), ",ondemand", ",spot")
+			if n, ok := az.Pricing[spotFeatures]; ok {
+				klog.V(4).Infof("Returning pricing for node %s: %+v from key %s", key, n, spotFeatures)
+				if key.GPUType() != "" {
+					n.Node.GPU = "1" // TODO: support multiple GPUs
+				}
+				return n.Node, nil
+			}
+			klog.V(1).Infof("[Info] found spot instance, trying to get retail price for %s: %s, ", spotFeatures, key)
+
+			spotCost, err := getRetailPrice(features[0], features[1], config.CurrencyCode)
+			if err == nil {
```

Comment:
Going with the comment above, propose we check `err != nil` here, log the error.

```go
if err != nil {
    // log error
} else {
    // continue expected code path
}
```